app/actions/TrezorActions.js

```diff
@@ -892,3 +911,299 @@ export const getWalletCreationMasterPubKey = () => async (
+      let addrValidResp = await wallet.validateAddress(walletService, inAddr);
+      const inAddr_n = addressPath(addrValidResp.getIndex(), 1, WALLET_ACCOUNT, chainParams.HDCoinType);
+      const commitAddr = decodedTicket.outputs[1].decodedScript.address;
+      addrValidResp = await wallet.validateAddress(walletService, commitAddr);
+      const commitAddr_n = addressPath(addrValidResp.getIndex(), 1, WALLET_ACCOUNT, chainParams.HDCoinType);
+      const ticketInput = {
+        address_n: inAddr_n,
+        prev_hash: decodedTicket.inputs[0].prevTxId,
+        prev_index: ticketOutN,
+        amount: decodedTicket.inputs[0].valueIn.toString()
+      };
+      const sstxsubmission = {
+          script_type: "SSTXSUBMISSIONPKH",
+          address: votingAddr,
+          amount: decodedTicket.outputs[0].value.toString()
+      };
+      // Trezor doesn't want opcodes included in op_return_data.
+      const ticketOPreturn = decodedTicket.outputs[1].script.slice(2).toString("hex");
+      const ticketsstxcommitment = {
+          script_type: "SSTXCOMMITMENTOWNED",
+          op_return_data: ticketOPreturn,
+          address_n: commitAddr_n,
+          amount: "0"
+      };
+      const ticketsstxchange = {
+          script_type: "SSTXCHANGE",
+          address: decodedTicket.outputs[2].decodedScript.address,
+          amount: "0"
+      };
+      const inputs = [ticketInput];
+      const outputs = [sstxsubmission, ticketsstxcommitment, ticketsstxchange];
+      const payload = await deviceRun(
+        dispatch,
+        getState,
+        async () => {
+          const res = await session.signTransaction({
+            coin: chainParams.trezorCoinName,
+            inputs: inputs,
+            outputs: outputs,
+            refTxs: [refTxs]
+          });
+          return res.payload;
+        }
+      );
+
+      const signedRaw = payload.serializedTx;
+      dispatch(publishTransactionAttempt(hexToBytes(signedRaw)));
+      // Pay fee.
+      console.log("waiting 5 seconds for the ticket to propogate throughout the network");
+      await new Promise(r => setTimeout(r, 5000));
+      const host = "https://" + vsp.host;
+      await payVSPFee(host, signedRaw, votingKey, accountNum.value, true, dispatch, getState);
+    }
+    dispatch({ type: TRZ_PURCHASETICKET_SUCCESS });
+  } catch (error) {
+    dispatch({ error, type: TRZ_PURCHASETICKET_FAILED });
+  }
+};
+
+function conToTrezCoinParams(coin) {
+  switch (coin) {
+    case "Decred":
+      return networks.decred;
+    case "Decred Testnet":
+      return networks.decredTest;
+    default:
+      throw "unsupported coin " + coin;
+  }
+}
+
+// payVSPFee attempts to contact a vst about a ticket and pay the fee if
+// necessary. It will search transacitons for a suitable fee transaction before
+// attempting to pay if newTicket is false.
+async function payVSPFee(host, txHex, votingKey, accountNum, newTicket, dispatch, getState) {
```

Comment:
This will need to be callable from somewhere else eventually. Will need to know the host though, and I'm not where to save that info. I suppose the user can choose per ticket where they think it came from if they still need to pay the fee for some reason. This will require ui changes which I am not comfortable doing in this pr...